packages/renderers-js/public/templates/fragments/programErrors.njk

```diff
@@ -26,3 +26,11 @@ export function {{ programGetErrorMessageFunction }}(code: {{ programErrorUnion
 
   return 'Error message not available in production bundles.';
 }
+
+export function {{ programIsErrorFunction }}<TProgramErrorCode extends {{ programErrorUnion }}>(
+    error: unknown,
+    transactionMessage: { instructions: Record<number, { programAddress: Address }> },
+    code?: TProgramErrorCode,
+): error is SolanaError<typeof SOLANA_ERROR__INSTRUCTION_ERROR__CUSTOM> & { context: { code: TProgramErrorCode } } {
```

Comment:
Maybe a dollop of `Readonly<>` on that return type?
```suggestion
): error is SolanaError<typeof SOLANA_ERROR__INSTRUCTION_ERROR__CUSTOM> & Readonly<{ context: Readonly<{ code: TProgramErrorCode }> }> {
```

Reply:
Good shout, we should probably do that in `@solana/programs` as well because that was copy/pasted from there.